Interleaved archive removal during downloads

s1ddok opened the pull request:
I was struggling to fit everything on my hard-drive so I thought this minor addition could be useful

mikeroberts3000 commented:
This is great, thank you. I just realized that the logic in this file is needlessly complicated, which is my fault. Can you please clean this file up a bit while we're making this change anyway? In particular, can you please:

- Make it so the `download` function only takes the URL as an input argument. All other arguments are constant across multiple function calls, so the download function can simply use the file-scope variables for `download_dir`, `decompress_dir`, and `remove_decompressed_archives`.
- Make it so there is not a redundant file-scope variable for `decompress_dir` and `remove_decompressed_archives`. It is cleaner to simply use `args.decompress_dir` and `args.remove_decompress_archives` instead.
- Rename `remove_decompressed_archives` to `remove_archive_after_decompress`, which I think conveys the behavior more clearly.

s1ddok commented:
Yes, I tried to follow the original code style but I think these ideas are actually good. I will incorporate them tomorrow and update the PR.

mikeroberts3000 commented:
Awesome! Thank you 😀

s1ddok commented:
okay so I've updated and tested the script. I've also introduced another addition: replaced 400+ invocations of download func with a single for loop over an array so that it is easier to introduce something like global progress lately

mikeroberts3000 commented on code/python/tools/dataset_download_images.py:
```diff
@@ -9,28 +9,20 @@
+parser.add_argument("--delete_archive_after_decompression", action="store_true")
 args = parser.parse_args()
 
-
```
For consistency with the rest of the codebase, can you please make it so there is one new-line here or three, but not two. Otherwise this code is ready to merge. Thank you 😀

mikeroberts3000 commented on code/python/tools/dataset_download_images.py:
```diff
@@ -9,28 +9,20 @@
-
 print("[HYPERSIM: DATASET_DOWNLOAD_IMAGES] Begin...")
 
-
```
See comment above.

s1ddok commented:
I've introduced 3-line-long break between logical steps and left one-line-long break between steps

mikeroberts3000 commented:
Thank you. There is still a two-line break starting on line 53 of your latest changes.

s1ddok commented:
ugh! fixed

mikeroberts3000 commented:
Great! Thank you 😀
